utils/knownAddresses.js

```diff
@@ -52,10 +54,54 @@ module.exports = function () {
 			return null;
 		};
 
+		this.setKnownAddress = (account) => {
+			if (!account || !account.address) return false;
+
+			let entry;
+			if (account.username) {
+				entry = { owner: account.username };
+			} else if (account.delegate && account.delegate.username) {
+				entry = { owner: account.delegate.username };
+			} else {
+				entry = { noOwner: true };
+			}
+
+			client.hmset(`address:${account.address}`, entry);
+
+			// Entry expires every day (86400 seconds) in case new delegate is registered
+			client.expire(`address:${account.address}`, 86400);
```

Comment:
The cache expires in order to check if a given address has been registered as delegate since last check. As delegate registration isn't an usual action, I thought 1 day to be ok.

One good idea would be update the cache when the delegate page is accessed in order to force the update even before the cache expires.

Reply:
A even better solution is to keep the cache up-to-date requesting regularly the latest transactions list with type "delegate registration"